spyder/api/folding.py

```diff
@@ -0,0 +1,365 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Copyright © <2013-2016> <Colin Duquesnoy and others, see pyqode/AUTHORS.rst>
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""
+This module contains the code folding API.
+Adapted from https://github.com/pyQode/pyqode.core/blob/master/pyqode/core/api/folding.py
+"""
+from __future__ import print_function
+import sys
+from spyder.utils.editor import TextBlockHelper
+
+
+def print_tree(editor, file=sys.stdout, print_blocks=False):
+    """
+    Prints the editor fold tree to stdout, for debugging purpose.
+
+    :param editor: CodeEditor instance.
+    :param file: file handle where the tree will be printed. Default is stdout.
+    :param print_blocks: True to print all blocks, False to only print blocks
+        that are fold triggers
```

Comment:
I think it's fine to leave this here so people can easily debug their detectors.